src/plugins/advanced_settings/public/component_registry/index.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { registerSettingsComponent, getSettingsComponent } from './component_registry';
+import {
+  registerDefaultComponents,
+  PAGE_TITLE_COMPONENT,
+  PAGE_SUBTITLE_COMPONENT,
+  PAGE_FOOTER_COMPONENT,
+} from './default_component_registry';
+
+registerDefaultComponents();
+
+export const componentRegistry = {
+  register: registerSettingsComponent,
+  get: getSettingsComponent,
+  componentType: {
+    PAGE_TITLE_COMPONENT,
+    PAGE_SUBTITLE_COMPONENT,
+    PAGE_FOOTER_COMPONENT,
+  },
```

Comment:
It feels like these should be static exports instead of part of the runtime contract, WDYT?

Perhaps in `public/index.ts` as:
```js
export const component = {
  PAGE_TITLE_COMPONENT,
  PAGE_SUBTITLE_COMPONENT,
  PAGE_FOOTER_COMPONENT,
};
```

Reply:
Seems like thats likely one more import. Changed to an enum which seems to make code a little more compact. Past that I don't have a strong opinion.